includes/admin/class-wp-job-manager-cpt.php

```diff
@@ -534,18 +533,57 @@ public function primary_column( $column, $screen ) {
 	}
 
 	/**
-	 * Removes all action links because WordPress add it to primary column.
-	 * Note: Removing all actions also remove mobile "Show more details" toggle button.
-	 * So the button need to be added manually in custom_columns callback for primary column.
+	 * Adds row actions for each job listing.
 	 *
 	 * @access public
-	 * @param array $actions
+	 * @param  array   $actions
+	 * @param  WP_Post $post
 	 * @return array
 	 */
-	public function row_actions( $actions ) {
+	public function row_actions( $actions, $post ) {
+
 		if ( 'job_listing' === get_post_type() ) {
-			return [];
+
+			unset( $actions['inline hide-if-no-js'] );
+			unset( $actions['trash'] );
+
+			if ( in_array( $post->post_status, [ 'pending', 'pending_payment' ], true ) && current_user_can( 'publish_post', $post->ID ) ) {
+				$admin_actions['approve'] = [
+					'action' => 'approved',
+					'name'   => __( 'Approve', 'wp-job-manager' ),
+					'url'    => wp_nonce_url( add_query_arg( 'approve_job', $post->ID ), 'approve_job' ),
+				];
+			}
+			if ( 'trash' !== $post->post_status ) {
+				if ( current_user_can( 'read_post', $post->ID ) ) {
+					$admin_actions['view'] = [
+						'action' => 'view',
+						'name'   => __( 'View', 'wp-job-manager' ),
+						'url'    => get_permalink( $post->ID ),
+					];
+				}
+				if ( current_user_can( 'edit_post', $post->ID ) ) {
+					$admin_actions['edit'] = [
+						'action' => 'edit',
+						'name'   => __( 'Edit', 'wp-job-manager' ),
+						'url'    => get_edit_post_link( $post->ID ),
+					];
+				}
+				if ( current_user_can( 'delete_post', $post->ID ) ) {
+					$admin_actions['delete'] = [
+						'action' => 'delete',
+						'name'   => __( 'Delete', 'wp-job-manager' ),
+						'url'    => get_delete_post_link( $post->ID ),
+					];
+				}
+			}
+			$admin_actions = apply_filters( 'job_manager_admin_actions', $admin_actions, $post );
```

Comment:
I tried to copy the use from the previous implementation so that it won't be different for other plugins that previous used the filters.  What do you think could happen in this case?

Reply:
Also, I think we should absolutely check with third-party devs ahead of this release.

Reply:
> What do you think could happen in this case?

I thought about weird things like having the links, and buttons as icons mixed. But maybe it's not a big deal. I'd just confirm if our official extensions aren't using it. And if some is using it, we could also update the HTML, if needed. :)

Reply:
The idea I had last night:  Once this is merged, I am going to ping our WPJM Friends in Slack and let them know about this coming change and see if they have questions or concerns.  How does that sound?

Reply:
Not sure if it's so important in order to make some type of "noise". I think it would be more important for us to check the plugins we maintain for now. I took the opportunity to run a search all in the plugins here for `job_manager_admin_actions`, and I didn't find anything. So I think we're fine here. :)